make static is unstable

## Flaky Test

### Which jobs are failing

```
gofmt ...
golangci-lint ...
level=info msg="[config_reader] Config search paths: [./ /home/runner/work/pd/pd/cmd/pd-server /home/runner/work/pd/pd/cmd /home/runner/work/pd/pd /home/runner/work/pd /home/runner/work /home/runner /home /]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[lintersdb] Active 15 linters: [bodyclose gocritic gosec gosimple govet ineffassign makezero misspell staticcheck stylecheck typecheck unconvert unparam unused whitespace]"
level=info msg="[loader] Go packages loading at mode 575 (types_sizes|compiled_files|exports_file|files|deps|imports|name) took 26.138491255s"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 308.028934ms"
panic: 295229738: duplicate method RunInTxn

goroutine 1419 [running]:
go/types.computeInterfaceTypeSet(0x0, 0xc000000006?, 0xc0048cc550)
	go/types/typeset.go:323 +0xb65
go/types.(*Interface).typeSet(...)
	go/types/interface.go:29
go/types.(*Interface).Complete(...)
	go/types/interface.go:143
golang.org/x/tools/internal/gcimporter.readUnifiedPackage(0xc00[195](https://github.com/tikv/pd/actions/runs/7217692652/job/19665912399#step:5:196)6000, 0x0, 0xc01ad62120, {0x1, 0x0, {0xc000f0bce0, 0x1e}, {0xc04d9036e0, 0x707d9}, {0xc00437e000, ...}, ...})
	golang.org/x/tools@v0.6.0/internal/gcimporter/ureader_yes.go:121 +0x4d8
golang.org/x/tools/internal/gcimporter.UImportData(0x133ee20?, 0xc00d704060?, {0xc04d4fc001?, 0x27?, 0xc01ad62120?}, {0xc000f0bce0, 0x1e})
	golang.org/x/tools@v0.6.0/internal/gcimporter/ureader_yes.go:68 +0x10a
golang.org/x/tools/go/gcexportdata.Read({0x133ee20?, 0xc00d704060?}, 0x0?, 0x7?, {0xc000f0bce0, 0x1e})
	golang.org/x/tools@v0.6.0/go/gcexportdata/gcexportdata.go:140 +0x270
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).loadFromExportData(0xc002538360?)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:221 +0x331
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).loadImportedPackageWithFacts(0xc002538360, 0x2)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:272 +0x4b
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).loadWithFacts(0xc000090000?, 0x43b396?)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:264 +0x137
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze(0xc002538360, 0x2, 0xc002375d40)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:61 +0xf4
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyzeRecursive.func1()
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:48 +0x1a8
sync.(*Once).doSlow(0x0?, 0x0?)
	sync/once.go:74 +0xc2
sync.(*Once).Do(...)
	sync/once.go:65
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyzeRecursive(0x0?, 0x0?, 0x0?)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:37 +0x52
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyzeRecursive.func1.1(0x0?)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:43 +0x2a
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyzeRecursive.func1
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:42 +0xae
make: *** [Makefile:176: static] Error 2
```


### CI link
https://github.com/tikv/pd/actions/runs/7208724232/job/19663480624?pr=7493
https://github.com/tikv/pd/actions/runs/7217692652/job/19665912399

### Reason for failure (if possible)


### Anything else

--- a/Makefile
+++ b/Makefile
@@ -160,7 +160,7 @@ SHELL := env PATH='$(PATH)' GOBIN='$(GO_TOOLS_BIN_PATH)' $(shell which bash)
 
 install-tools:
 	@mkdir -p $(GO_TOOLS_BIN_PATH)
-	@which golangci-lint >/dev/null 2>&1 || curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GO_TOOLS_BIN_PATH) v1.51.2
+	@which golangci-lint >/dev/null 2>&1 || curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GO_TOOLS_BIN_PATH) v1.55.2
 	@grep '_' tools.go | sed 's/"//g' | awk '{print $$2}' | xargs go install
 
 .PHONY: install-tools

